proc.c

```diff
@@ -1572,12 +1573,22 @@ mnew_internal(const rb_method_entry_t *me, VALUE klass, VALUE iclass,
     }
     if (me->def->type == VM_METHOD_TYPE_ZSUPER) {
 	if (me->defined_class) {
-	    VALUE klass = RCLASS_SUPER(RCLASS_ORIGIN(me->defined_class));
+            klass = RCLASS_SUPER(RCLASS_ORIGIN(me->defined_class));
 	    id = me->def->original_id;
             me = (rb_method_entry_t *)rb_callable_method_entry_with_refinements(klass, id, &iclass);
 	}
 	else {
-	    VALUE klass = RCLASS_SUPER(me->owner);
```

Comment:
Doesn't it work just by taking out the ORIGIN here?

Reply:
I thought I tried that, but maybe I did `RCLASS_ORIGIN(RCLASS_SUPER(me->owner))` instead of `RCLASS_SUPER(RCLASS_ORIGIN(me->owner))`.  I can see if one of those works.

Reply:
Looks like that `RCLASS_SUPER(RCLASS_ORIGIN(me->owner))`  works, pushed commit with that.